### DISCUSSION TITLE
[3.0.next] peer deps broken in gatsby-recipes

### DISCUSSION THREAD

**Discussion by moyarich** [gatsbyjs/gatsby#30122] [Help]
Got dependency error while installing [gatsby-cli](https://github.com/gatsbyjs/gatsby)

`npm install -g gatsby-cli`

# Error :
 Could not resolve dependency: peer graphql@"^14.4.1" from express-graphql@0.9.0

Could not resolve [graphql](https://github.com/graphql/graphql-js) dependency because [gatsby-recipes](https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-recipes) is using express-graphql@"^0.9.0"

```bash
  ├─ gatsby-cli@3.0.0
     └─ gatsby-recipes@0.11.0
        └─ express-graphql@^0.9.0
           └─ graphql@^14.4.1
```

*I am using [Yarn resolution](https://www.gatsbyjs.com/docs/reference/release-notes/migrating-from-v2-to-v3/#handling-dependencies-for-plugins-that-are-not-yet-updated) to address my dependency issues, but below is a suggestion on how this issues can be fixed in gatsby-recipes.*

# Suggestion - how to fix error: 
the newest version of graphql is 15.5.0 and is supported by [express-graphql](https://github.com/graphql/express-graphql)@[0.12.0](https://semver.npmjs.com/).

**My  `gatsby info --clipboard` output :**
```
 System:
    OS: macOS 11.1
    CPU: (4) x64 Intel(R) Core(TM) i5-5257U CPU @ 2.70GHz
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 15.11.0 - /usr/local/bin/node
    Yarn: 1.22.10 - /usr/local/bin/yarn
    npm: 7.6.0 - /usr/local/bin/npm
  Languages:
    Python: 2.7.16 - /usr/bin/python
  Browsers:
    Chrome: 88.0.4324.192
    Firefox: 86.0
    Safari: 14.0.2
  npmGlobalPackages:
    gatsby-cli: 3.0.0
```

**Comment by LekoArts** [comment]
Hi!

npm 7 is unfortunately a bit more unforgiving about such things, so if you run it with the `--legacy-peer-deps` flag the installation will work. Bumping the dependency is possibly dangerous as they don't have a changelog and don't use semantic versioning (pre v1).
